cmd/admin-handlers.go

```diff
@@ -979,6 +980,28 @@ func (adminAPI adminAPIHandlers) SetConfigHandler(w http.ResponseWriter, r *http
 		return
 	}
 
+	var config serverConfigV19
```

Comment:
Thought (not to be treated as a suggestion necessarily):
Committing to a concrete type like `serverConfigV19` puts `pkg/madmin` API handler in a spot every time server configuration type is modified (version19 tells us something :) ). To get out of this situation we could do,
```go
       var creds struct {
		Credential struct {
			AccessKey string `json:"accessKey"`
			SecretKey string `json:"secretKey"`
		} `json:"credential"`
	}
	err := json.Unmarshal(configBytes, &creds)
	if err != nil {
		// error handling goes here
	}

```

The above snippet focuses only on that portion of the server configuration type that we require to validate here. Note: using `serverConfigV19` doesn't provide any type-safety since `pkg/madmin` and server reside in different packages.